caddyhttp/proxy/upstream.go

```diff
@@ -150,7 +150,8 @@ func (u *staticUpstream) From() string {
 
 func (u *staticUpstream) NewHost(host string) (*UpstreamHost, error) {
 	if !strings.HasPrefix(host, "http") &&
-		!strings.HasPrefix(host, "unix:") {
+		!strings.HasPrefix(host, "unix:") &&
+		!strings.HasPrefix(host, "quic:") {
```

Comment:
It can.
But one of the features of QUIC is the 1- or 0-RTT connection setup and you're losing that if you set up a connection over TCP first. So I would really keep it there. 
Besides, I don't know if Caddy supports the Alt-Svc headers as a client?

This is also what Google does inside Chrome for it's own QUIC-enabled servers, directly connecting over QUIC.